src/setuptools_scm/version.py

```diff
@@ -132,6 +132,7 @@ def __init__(
         dirty=False,
         preformatted=False,
         branch=None,
+        config=None,
```

Comment:
i am on the fence on whether configuration should be a parameter to versions

Reply:
That's fair enough, but if the branch name is to be parsed it needs config from somewhere - I was having it use the default config, but this way it can honour version regexes set by the user, I guess. Happy to have it just use the default config though!

Reply:
i understand the necessity - i'm just wondering about how to pass in the config
it must land in the version formatter/generator somehow

and its also going to be useful for a semver-towncrier scheme, that will pick versions number based on 

i'll sleep a night over this

both the pass in as part of the version object as well as adding it as extra parameter to the version formatter have different ups and downs